### DISCUSSION TITLE
_queue.Empty Error when testing Module 6 program

### DISCUSSION THREAD

**Discussion by chasedamagician** [AccelerationConsortium/ac-microcourses#26] [Hello World]
Hello @sgbaird , I am finishing up the Module 6 assignment but when I run the pytest command, I get an error stating that 
"_queue.Empty: Did not receive any data on topic <course_id>/as7341 after publishing to <course_id>/neopixel"
I checked the queue and confirmed that the dictionary is being added to the queue and there is data being published from the topic <course_id>/as7341. What could the issue be? Thank you.

here is the link to my code:
https://github.com/ACC-HelloWorld/6-connecting-the-pieces-chasedamagician

**Comment by sgbaird** [comment]
> Hello @sgbaird , I am finishing up the Module 6 assignment but when I run the pytest command, I get an error stating that 
> "_queue.Empty: Did not receive any data on topic <course_id>/as7341 after publishing to <course_id>/neopixel"
> I checked the queue and confirmed that the dictionary is being added to the queue and there is data being published from the topic <course_id>/as7341. What could the issue be? Thank you.
> 
> here is the link to my code:
> https://github.com/ACC-HelloWorld/6-connecting-the-pieces-chasedamagician
> 
> 

Are you seeing the sensor data being passed in the HiveMQ broker? You can check using the interactive cloud console attached to your broker. It's one of the tabs along the top when you're logged in and click on the cluster. 

If you run orchestrator.py directly while microcontroller.py is running on the Pico W, does it complete successfully to the end, or do you get the same error? Are you able to run multiple iterations of the light mixing experiments?

**Reply by chasedamagician** [reply]
Yes the data is being passed to the broker, I can see that the program is publishing both from the neopixel topic and the as7341 topic, and I do not get this error if I directly run the microcontroller program on the Pico W. It only appears when I use pytest.

**Reply by sgbaird** [reply]
If the only difference is the use of pytest vs. running orchestrator.py directly, and the pytest error occurs before any light-mixing experiments start, it is likely an issue I'll need to debug. One last thing - do you mind running microcontroller.py and then triggering the autograding? (E.g., add an extra space to a file), then include the link to the GitHub actions workflow here?

This would help me to reference the full example.

**Reply by chasedamagician** [reply]
Hello, this is the link to the action workflow, Thank you.
https://github.com/ACC-HelloWorld/6-connecting-the-pieces-chasedamagician/actions/runs/10048746023/job/27773361040

**Reply by sgbaird** [reply]
You'll want to focus on the error from `credentials_test.py`:

```
E           TimeoutError: Double check that your HiveMQ credentials are correct (host, username, password), and that the username/password combination has both subscribe and publish permissions. Refer to the README for instructions.
  
  credentials_test.py:57: TimeoutError
```

An error during this test indicates that either something is off with the way the codespaces secrets are set up or how the HiveMQ broker itself is set up.

You can troubleshoot by replacing your credentials with the public test credentials from module 4.

**Reply by chasedamagician** [reply]
Solved, it was an issue with the microcontroller_test.py, the pay_dict that it uses does contain the session_id which led to a KeyError when you try to test the code.
